src/main/java/apoc/cypher/Cypher.java

```diff
@@ -365,4 +365,56 @@ private T poll() {
         Result result = db.execute(cypher, params == null ? Collections.EMPTY_MAP : params);
         return result.stream().map(stringObjectMap -> new MapResult(stringObjectMap));
     }
+
+    @Procedure("apoc.when")
+    @Description("apoc.when(condition, ifQuery, elseQuery:'', params:{}) yield value - based on the conditional, executes read-only ifQuery or elseQuery with the given parameters")
+    public Stream<MapResult> when(@Name("condition") boolean condition, @Name("ifQuery") String ifQuery, @Name(value="elseQuery", defaultValue = "") String elseQuery, @Name(value="params", defaultValue = "") Map<String, Object> params) {
+        if (params == null) params = Collections.emptyMap();
+        String targetQuery = condition ? ifQuery : elseQuery;
+
+        if (targetQuery.isEmpty()) {
+            return Stream.of(new MapResult(Collections.emptyMap()));
+        } else {
+            return db.execute(withParamMapping(targetQuery, params.keySet()), params).stream().map(MapResult::new);
+        }
+    }
+
+    @Procedure(value="apoc.when.do", mode = Mode.WRITE)
+    @Description("apoc.when.do(condition, ifQuery, elseQuery:'', params:{}) yield value - based on the conditional, executes writing ifQuery or elseQuery with the given parameters")
+    public Stream<MapResult> whenDo(@Name("condition") boolean condition, @Name("ifQuery") String ifQuery, @Name(value="elseQuery", defaultValue = "") String elseQuery, @Name(value="params", defaultValue = "") Map<String, Object> params) {
+        return when(condition, ifQuery, elseQuery, params);
+    }
+
+    @Procedure("apoc.whenMany")
+    @Description("apoc.whenMany([[condition, query], [condition, query], ...], elseQuery:'', params:{}) yield value - given conditional / read-only query pairs, executes the query associated with the first conditional evaluating to true (or the else query if none are true) with the given parameters")
+    public Stream<MapResult> whenMany(@Name("conditionals") List<List<Object>> conditionals, @Name(value="elseQuery", defaultValue = "") String elseQuery, @Name(value="params", defaultValue = "") Map<String, Object> params) {
```

Comment:
Shouldn't we just call it apoc.case ? Which takes either this list or functions that wrap condition + query ?
like a function `apoc.pred(condition, query)` 
the other thing we can do is to use defaults, 

i.e. `apoc.case(Boolean pred1, String query1, Boolean pred2=null, String query2=null,....) `
so it looks like varargs from the outside

another function we should add is `apoc.args` which turns a varargs into a list.

`List<Object> apoc.args(Object=null value1, Object=null value2, Object=null value3, Object=null value3, ..., Object=null value10, ...)`